### PR TITLE
Oxt580 syslinux mobile broadband

### PR DESCRIPTION
**Pull request by cdkell3**
[OXT-580] Updated syslinux and mobile-broadband-provider-info to rely more on upstream OE.

syslinux was moved to a bbappend that include the few things we required that OE did not and moves Xen up to the newer version OE is using.

mobile-broadband-provider-info was moved to a bbappend and while its unchanged this will allow Xen to use any changes upstream OE makes in the future.

### CODE REVIEW

**Comment by eric-ch** [comment]
Could you rebase on master to clarify the history? You have two commits afaict.
Also can you remove the trailing empty lines from the bbappend?
I would also think you can append syslinux_6.%.bbappend as your append should not change for minor upgrades. Same for acpid_2.0.%.bbappend (maybe 2.%.bbappend even).
Thanks!

File: recipes-devtools/syslinux/syslinux_6.03.bbappend [hunk 1]
```diff
@@ -0,0 +1,14 @@
+PACKAGES =+ "${PN}-isohybrid"
+PACKAGES =+ "${PN}-mboot"
```
Comment (eric-ch) [inline recipes-devtools/syslinux/syslinux_6.03.bbappend]: Other recipes tend to use the following stanza:
```sh
PACKAGES =+ " \
    ${PN}-isohybrid \
    ${PN}-mboot \
"
```

File: recipes-bsp/acpid/acpid_2.0.23.bbappend [hunk 2]
```diff
@@ -0,0 +1,28 @@
+FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"
+
+
+SRC_URI = "${SOURCEFORGE_MIRROR}/acpid2/acpid-${PV}.tar.xz \
+           file://init \
+           file://powerbtn \
+           file://ac \
+           file://ac_actions \
+	   file://acpid.service \
```
Comment (eric-ch) [inline recipes-bsp/acpid/acpid_2.0.23.bbappend]: Indent skew.

**Comment by cdkell3** [comment]
I have no idea how the acpid ended up in here. I did rebase off master then made this branch. Those changes have already been accepted.

File: recipes-devtools/syslinux/syslinux_6.%.bbappend [hunk 3]
```diff
@@ -1,5 +1,6 @@
-PACKAGES =+ "${PN}-isohybrid"
-PACKAGES =+ "${PN}-mboot"
+PACKAGES =+ " \
+	     ${PN}-isohybrid \"
+             ${PN}-mboot \"
```
Comment (eric-ch) [inline recipes-devtools/syslinux/syslinux_6.%.bbappend]: Something went wrong in this change.

**Comment by eric-ch** [comment]
The history is getting confusing, I had a quick look with your branch:
```sh
$ git rebase master
```
Will have one conflict with recipes-devtools/syslinux/syslinux_6.%.bbappend. Actually that will give the opportunity to amend https://github.com/OpenXT/xenclient-oe/pull/546/commits/1620e4c7ef97d9b688d85fd3f13893e59e5e536b (see inline comment).
```sh
$ git add recipes-devtools/syslinux/syslinux_6.%.bbappend
$ git rebase --continue
$ git log --oneline master..HEAD
a30441a Deleted old 6.03 bbappend
ef75668 Renamed file to wildcard minor versions
b54baf1 [OXT580] Updated mobile-broadband-provider-info and syslinux to use upstream OE.
```
I believe these are the changes you intended to PR? The history is looks more clean (without merge commits & all).

File: recipes-devtools/syslinux/syslinux_6.%.bbappend (hunk 3 above)
Comment (cdkell3) [inline recipes-devtools/syslinux/syslinux_6.%.bbappend]: I'm not sure what you mean? As requested I removed the two PACKAGES=+ line and made one PACKAGE=+ over multiple lines with escapes.

**Comment by cdkell3** [comment]
That should be it. If it makes the history cleaner that works for me.

File: recipes-devtools/syslinux/syslinux_6.%.bbappend (hunk 3 above)
Comment (eric-ch) [inline recipes-devtools/syslinux/syslinux_6.%.bbappend]: Bitbake will not parse this:
```
ERROR: ParseError at [...]xenclient-oe/recipes-devtools/syslinux/syslinux_6.%.bbappend:3: unparsed line: '             ${PN}-mboot \"'
```
```PACKAGES =+``` does not have an ending double-quote and there is an escaped one in between which should not be here. The snippet provided earlier was:
```sh
PACKAGES =+ " \
    ${PN}-isohybrid \
    ${PN}-mboot \
"
```

Comment (cdkell3) [inline recipes-devtools/syslinux/syslinux_6.%.bbappend]: Ah, I see now. My apologies.

**Comment by eric-ch** [comment]
ping?

**Comment by cdkell3** [comment]
Oops, I completely misunderstood what you were doing. I thought you'd managed to clean up my awful git history and commit it. I'm trying this now but I'm not sure it worked as intended. Our environment here is a bit weird. If I ever do another PR, I'll be starting from a fresh checkout of the master here on github, rather than the local one we have.

**Comment by eric-ch** [comment]
@cdkell3 It did not :). I rebased again, ending up with only one commit (squashed the renaming with the initial change) : https://github.com/eric-ch/xenclient-oe/tree/cdkell-oxt-580-syslinux-mobile-broadband. If you are ok with that, I will open a different PR, from branch with your commit, and close this one.

**Comment by cdkell3** [comment]
That's fine. Thank you.

**Comment by eric-ch** [comment]
See https://github.com/OpenXT/xenclient-oe/pull/592 for the new PR.

**Pull request closed without merging**